src/include/tiered.h

```diff
@@ -46,6 +46,24 @@ struct __wt_tiered_manager {
 #define WT_TIERED_NAME_SHARED 0x8u
 /* AUTOMATIC FLAG VALUE GENERATION STOP */
 
+/* Flush tier flags */
+/* AUTOMATIC FLAG VALUE GENERATION START */
+#define WT_FLUSH_TIER_FORCE 0x1u
+#define WT_FLUSH_TIER_OFF 0x2u
+#define WT_FLUSH_TIER_ON 0x4u
+/* AUTOMATIC FLAG VALUE GENERATION STOP */
+
+/*
+ * The flush state is part of the connection. It is a 64-bit unsigned integer with two 32-bit
+ * components. There is an addition counter and a decrement counter. A flush tier call will add work
+ * units to the increment counter and the internal thread processing them will add to the decrement
+ * counter. When the two parts are the same then the processing is done.
+ */
+#define WT_FLUSH_STATE_ADD(state) ((uint64_t)(uint32_t)(state))
```

Comment:
This looks weird with two casts.  If the intent is to mask out the upper 32 bits, it's probably better using bit mask operations.

But maybe better is breaking up flush_state into a struct with two 32 bit ints.  Then the accesses won't need macros or bit fiddling.  I see that flush_state wants to be 64 bits for a CAS instruction.  Then I would make a union of 64 bit int and a struct pair of 32 bit ints.

Reply:
TBH I copied this almost exactly from how we manipulate the state in the log slots for joining and releasing bytes in there, `log.h`. Instead of breaking it up into how many we added and how many we processed, I could just have a counter that gets incremented and decremented as a single number. It should never go below 0 and could really be just 32 bits, atomically used.

Reply:
I made it a simple 32 bit atomic counter. Fixed.